### PR TITLE
FixerConfigurationResolver::addOption - remove method

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

File: src/Fixer/Alias/NoMixedEchoPrintFixer.php [hunk 1]
```diff
@@ -63,17 +63,15 @@ public function configure(array $configuration = null)
...
         $use = new FixerOption('use', 'The desired language construct.');
         $use
             ->setAllowedValues(array('print', 'echo'))
             ->setDefault('echo')
         ;
 
-        return $configurationDefinition
-            ->addOption($use)
-        ;
+        return new FixerConfigurationResolver(array(
```
Comment (julienfalque) [inline src/Fixer/Alias/NoMixedEchoPrintFixer.php:72]: Maybe better on a single line when there are few options?

File: src/FixerConfiguration/FixerConfigurationResolver.php [hunk 2]
```diff
@@ -116,4 +113,25 @@ public function resolve(array $options)
...
     }
+
+    /**
+     * @param FixerOptionInterface $option
+     *
+     * @throws \LogicException when the option is already defined
+     *
+     * @return $this
+     */
+    private function addOption(FixerOptionInterface $option)
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolver.php:128]: This should be moved into the constructor directly, as it's only used there.

File: src/FixerConfiguration/FixerConfigurationResolver.php [hunk 3]
```diff
@@ -116,4 +113,25 @@ public function resolve(array $options)
...
+    {
+        $name = $option->getName();
+
+        if (in_array($name, $this->registeredNames, true)) {
+            throw new \LogicException(sprintf('The "%s" option is already defined.', $name));
```
Comment (julienfalque) [inline src/FixerConfiguration/FixerConfigurationResolver.php]:
```diff
-already defined
+defined multiple times
```

File: tests/FixerConfiguration/FixerConfigurationResolverTest.php [hunk 4]
```diff
@@ -24,39 +24,30 @@
 {
     public function testAddOption()
```
Comment (julienfalque) [inline tests/FixerConfiguration/FixerConfigurationResolverTest.php]: Test should be renamed to something like `testWithDuplicatedOptions`.

**Review by julienfalque — CHANGES_REQUESTED** [review]
Should we throw an exception when the options array is empty?

File: src/Fixer/Alias/NoMixedEchoPrintFixer.php (hunk 1 above)
Comment (keradus) [reply src/Fixer/Alias/NoMixedEchoPrintFixer.php:72]: I don't care, I applied the change automatically

File: src/FixerConfiguration/FixerConfigurationResolver.php (hunk 3 above)
Comment (keradus) [reply src/FixerConfiguration/FixerConfigurationResolver.php]: :+1:

File: tests/FixerConfiguration/FixerConfigurationResolverTest.php (hunk 4 above)
Comment (keradus) [reply tests/FixerConfiguration/FixerConfigurationResolverTest.php]: :+1:

File: src/FixerConfiguration/FixerConfigurationResolver.php (hunk 2 above)
Comment (keradus) [reply src/FixerConfiguration/FixerConfigurationResolver.php:128]: I prefer to have this method helper, so it automatically check `FixerOptionInterace` type on syntax level.

**Comment by keradus** [comment]
done

**Review by julienfalque — APPROVED** [review]

**Pull request merged**